Add test module

MuhammadAbdullah191 opened the pull request:
- added functionality for admin to add question and their options
- added functionality for regular user will be able to take personality-test
- added functionality of dynamic screens for personality test where user will be able to navigate through questions one by one

Commit:
Solved webpacker problem and added devise for admin authentication

Commit:
added admin ability to add questions

Commit:
added ability to take personality test

Commit:
applied rubocop

Commit:
fix path bugs

husnainrazadevsinc commented on app/controllers/admin/questions_controller.rb:
```diff
@@ -0,0 +1,58 @@
+
+    def new
+      @question = Question.new
+      4.times { @question.options.build }
```
Please use Global constant "NO_OF_OPTIONS" instead of 4 here

MuhammadAbdullah191 replied:
sure

husnainrazadevsinc commented on app/views/admin/questions/_form.html.erb:
```diff
@@ -0,0 +1,33 @@
+              <% end %>
+            </ul>
+          </div>
+        <% end %>
```
I think there is no need to display these errors here in the form, the piece of code for flash in application.html.erb file should be enough, please check

MuhammadAbdullah191 replied:
noted

husnainrazadevsinc commented on app/views/home/index.html.erb:
```diff
@@ -0,0 +1,70 @@
+        <p><%= question.title %></p>
+        <% question.options.each do |option| %>
+        <label class="d-block">
+          <%= radio_button_tag("question_#{question.id}", option.id) %> <%=
```
please make a helper for that string

MuhammadAbdullah191 replied:
done

husnainrazadevsinc commented on app/views/home/index.html.erb:
```diff
@@ -0,0 +1,70 @@
+    </div>
+    <% end %> <% end %>
+  </div>
+</div>
```
spacing issues... please check

MuhammadAbdullah191 replied:
sure

husnainrazadevsinc commented on app/views/home/index.html.erb:
```diff
@@ -0,0 +1,70 @@
+      }
+    });
+  });
+</script>
```
kindly add this script code in a separate javascript file

MuhammadAbdullah191 replied:
noted

husnainrazadevsinc commented on app/views/layouts/application.html.erb:
```diff
@@ -4,12 +4,22 @@
+<% end %>
+<% if alert %>
+  <p class="alert alert-danger"><%= alert %></p>
+<% end %>
```
please make a partial for it

MuhammadAbdullah191 replied:
done

husnainrazadevsinc commented on config/routes.rb:
```diff
@@ -1,3 +1,16 @@
+  end
+  authenticated :admin do
+    root 'admin/questions#index', as: :authenticated_root
+  end
```
please try use resourceful routes

MuhammadAbdullah191 replied:
noted

husnainrazadevsinc commented on app/controllers/home_controller.rb:
```diff
@@ -0,0 +1,34 @@
+    if question_values.length < count
+      redirect_to root_path, alert: 'Please answer all questions'
+      return
+    end
```
please put this check in another private method

MuhammadAbdullah191 replied:
sure

husnainrazadevsinc commented on app/controllers/home_controller.rb:
```diff
@@ -0,0 +1,34 @@
+    question_values.each do |n|
+      id = n.to_i
+      option_type = Option.find(id).option_type
+      result = if option_type == 'introvert'
```
please use enum helpers here instead of comparing strings.

MuhammadAbdullah191 replied:
done

husnainrazadevsinc reviewed (changes requested):
I have given feedback. Please check 👍

ABDULLAHBADAR committed:
Application: Addressed Feedback
-> Added Global Constant
-> QuestionFormView: Remove extra error block
-> Made Helper for question no
-> Solved spacing issues
-> Separated JS script from erb file
-> Made flash messages partial
-> Used Resourceful routes
-> Used enum helper instead of comparing string

husnainrazadevsinc commented on app/controllers/admin/questions_controller.rb:
```diff
@@ -1,58 +1,57 @@
+    if @question.save
+      redirect_to admin_questions_path, notice: 'Question created successfully.'
+    else
+      render :new
```
please render error status along with error messages

husnainrazadevsinc commented on app/controllers/admin/questions_controller.rb:
```diff
@@ -1,58 +1,57 @@
+    if @question.update(question_params)
+      redirect_to admin_questions_path, notice: 'Question updated successfully.'
+    else
+      render :edit
```
same as above!

husnainrazadevsinc commented on app/helpers/home_helper.rb:
```diff
@@ -1,2 +1,5 @@
 module HomeHelper
+	def questionNo(q)
+		"question_#{q.id}"
+	end
```
spacing issues

MuhammadAbdullah191 replied:
there is no spacing issue on my side, it might be due to something else, let me check

husnainrazadevsinc commented on app/javascript/packs/dynamic_form.js:
```diff
@@ -0,0 +1,33 @@
+		} else {
+			$(".submit-btn").addClass("disabled");
+		}
+	});
```
spacing issues

husnainrazadevsinc commented on app/views/home/index.html.erb:
```diff
@@ -3,68 +3,32 @@
-    });
-  });
-</script>
+<%= javascript_pack_tag 'dynamic_form' %>
```
why did you use javascript tag?
please require the required js file in the application.js file

MuhammadAbdullah191 replied:
ok, noted

husnainrazadevsinc reviewed (changes requested):
Suggested some changes. Please check

ABDULLAHBADAR committed:
Application: Addressed Feedback
-> Refactored Code

husnainrazadevsinc commented on app/javascript/packs/dynamic_form.js:
```diff
@@ -1,33 +1,30 @@
+		$('input[type="radio"]').click(function () {
+			let checkedAnswers = $("input[type=radio]:checked").length;
+			let count = parseInt(document.getElementsByClassName('step').length)
+			if (count === checkedAnswers) {
```
I think there is some issue with your tab spacing

ABDULLAHBADAR committed:
Helper: Added helper for finding last question
Removed Extra javascript tag
